API/Controllers/PortfolioController.cs

```diff
@@ -0,0 +1,498 @@
+        [ProducesResponseType(typeof(PortfolioResourceResult), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.Unauthorized)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.NotFound)]
+        public async Task<IActionResult> UpdatePortfolio(int portfolioId, [FromBody] PortfolioResource portfolioResource)
+        {
+            User user = await HttpContext.GetContextUser(userService).ConfigureAwait(false);
+
+            Portfolio portfolio = await portfolioService.FindAsync(portfolioId).ConfigureAwait(false);
+            if(portfolio == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to update portfolio.",
+                    Detail = "The specified portfolio could not be found in the database.",
+                    Instance = "65229A71-FDD8-4887-BBCD-3D4258C5D329"
+                };
+                return NotFound(problem);
+            }
+
+
+
+            if(!(portfolio.User.Id == user.Id))
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed to edit the portfolio.",
+                    Detail = "The user is not allowed to edit the portfolio.",
+                    Instance = "B00CA783-B8E0-4F73-9E19-A74448F1930B"
+                };
+                return Unauthorized(problem);
+            }
+
+            mapper.Map(portfolioResource, portfolio);
+            portfolioService.Update(portfolio);
+            portfolioService.Save();
+            return Ok(mapper.Map<Portfolio, PortfolioResourceResult>(portfolio));
+        }
+
+        /// <summary>
+        /// This method is responsible for deleting the portfolio.
+        /// </summary>
+        /// <param name="portfolioId">The portfolio identifier which is used for searching the portfolio.</param>
+        /// <returns>This method returns status code 200.</returns>
+        /// <response code="200">This endpoint returns status code 200. The current portfolio is deleted.</response>
+        /// <response code="404">The 404 Not Found status code is returned when the current portfolio could not be found.</response>
+        [HttpDelete]
+        [Authorize]
+        [ProducesResponseType(typeof(PortfolioItemResourceResult), (int) HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.Unauthorized)]
+        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.NotFound)]
+        public async Task<IActionResult> DeletePortfolio(int portfolioId)
+        {
+            Portfolio portfolio = await portfolioService.FindAsync(portfolioId).ConfigureAwait(false);
+            User user = await HttpContext.GetContextUser(userService).ConfigureAwait(false);
+
+            if(portfolio.User.Id != user.Id)
+                return Forbid();
+
+
+            if(portfolio == null)
```

Comment:
The way you structured this code, it not completely correct. The check in the if statement (line 233) always returns false. The reason is: line 229 tries to access a property in portfolio. So if this would be null, it already throws an error. I hope this makes sense :) (please let me know if it does not).

So a solution could be to switch the order of both checks.

Reply:
I changed the postion of those two. I Also deleted the check if user id is the portfolio user id. Because it double checked the same parameters